packages/blade/src/components/Collapsible/Collapsible.tsx

```diff
@@ -79,24 +101,41 @@ const Collapsible = ({
       onExpandChange: handleExpandChange,
       defaultIsExpanded: initialDefaultExpanded.current,
       direction,
+      collapsibleBodyId,
     }),
-    [isBodyExpanded, direction, handleExpandChange, isExpanded],
+    [isBodyExpanded, direction, handleExpandChange, isExpanded, collapsibleBodyId],
   );
 
+  Children.forEach(children, (child) => {
+    if (
+      !(
+        isValidAllowedChildren(child, MetaConstants.CollapsibleBody) ||
+        isValidAllowedChildren(child, MetaConstants.CollapsibleButton) ||
+        isValidAllowedChildren(child, MetaConstants.CollapsibleLink) ||
+        isValidAllowedChildren(child, MetaConstants.AccordionButton)
+      )
+    ) {
+      console.warn(
```

Comment:
probably throw?

Reply:
Let's finalize based on https://github.com/razorpay/blade/pull/1285#discussion_r1229028226. I don't have strong opinions but leaning towards not throwing.

Reply:
Do this in an effect on mount?

Reply:
On other components we're doing this on each render 🤔 

I think ideally these checks would be moved to dev only #272